tests/Aspire.Hosting.SqlServer.Tests/SqlServerFunctionalTests.cs

```diff
@@ -0,0 +1,284 @@
+
+        var sqlserver = builder.AddSqlServer("sqlserver");
+        var testDb = sqlserver.AddDatabase(TestDbName);
+        var masterDb = sqlserver.AddDatabase(MasterDbName);
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{testDb.Resource.Name}"] = await testDb.Resource.ConnectionStringExpression.GetValueAsync(default),
+            [$"ConnectionStrings:{masterDb.Resource.Name}"] = await masterDb.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddKeyedSqlServerClient(masterDb.Resource.Name);
+        hb.AddSqlServerClient(testDb.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var masterDbConnection = host.Services.GetRequiredKeyedService<SqlConnection>(masterDb.Resource.Name);
+            var testDbConnection = host.Services.GetRequiredService<SqlConnection>();
+
+            await CreateTestDb(masterDbConnection, testDbConnection, token);
+
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [SkipOnCI("https://github.com/dotnet/aspire/issues/5055")]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+
+            var password = "P@ssw0rd1";
+
+            var passwordParameter = builder1.AddParameter("pwd");
+            builder1.Configuration["Parameters:pwd"] = password;
+
+            var sqlserver1 = builder1.AddSqlServer("sqlserver", passwordParameter);
+            var testdb1 = sqlserver1.AddDatabase(TestDbName);
+            var masterDb1 = sqlserver1.AddDatabase(MasterDbName);
+
+            if (useVolume)
+            {
+                // Use a deterministic volume name to prevent them from exhausting the machines if deletion fails
+                volumeName = VolumeNameGenerator.CreateVolumeName(sqlserver1, nameof(WithDataShouldPersistStateBetweenUsages));
+
+                // If the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                sqlserver1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Directory.CreateTempSubdirectory().FullName;
```

Comment:
The permissions issue might be due to this method. Try creating a standard temporary folder for the cases when this one fails. This method explicitly removes some permissions which might be necessary in this case.

Reply:
I did, But still not working.